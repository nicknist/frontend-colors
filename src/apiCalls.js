export const getProjects = () => {
  return fetch('http://backend-colors.herokuapp.com/api/v1/projects')
    .then(response => {
      if (!response.ok) {
        throw Error('Can not find any projects!')
      }
      return response.json();
    })
}

export const getProject = (id) => {
  return fetch(`http://backend-colors.herokuapp.com/api/v1/projects/${id}`)
    .then(response => {
      if(!response.ok) {
        throw Error('Can not find Palette');
      }
      return response.json();
    })
}

export const getPalettes = () => {
  return fetch('http://backend-colors.herokuapp.com/api/v1/palettes')
    .then(response => {
      if (!response.ok) {
        throw Error('Can not find any projects!')
      }
      return response.json();
    })
}

export const getPalette = (id) => {
  return fetch(`http://backend-colors.herokuapp.com/api/v1/palettes/${id}`)
    .then(response => {
      if(!response.ok) {
        throw Error('Can not find Palette');
      }
      return response.json();
    })
}
