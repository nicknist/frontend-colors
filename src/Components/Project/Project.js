import React, { useState, useEffect } from 'react';
import Palette from '../Palette/Palette';
import { deleteProject } from '../../apiCalls';
import { getPalettes } from '../../helperFunctions';

const Project = ({ info }) => {
  const [palettes, setPalettes] = useState([]);

  useEffect(() => {
    let palettes = [];
    if (info.palette1_id) palettes.push(info.palette1_id);
    if (info.palette2_id) palettes.push(info.palette2_id);
    if (info.palette3_id) palettes.push(info.palette3_id);
    getPalettes(palettes).then(data => setPalettes(data));
  })

  return (
    <>
      <h1>{info.title}</h1>
      <button className='randomizer delete-project' onClick={() => deleteProject(info.id)}>NUKE IT</button>
      {palettes[0] ? <Palette
        key={palettes[0].color1}
        name={palettes[0].title}
        colors={[
          {color: palettes[0].color1},
          {color: palettes[0].color2},
          {color: palettes[0].color3},
          {color: palettes[0].color4},
          {color: palettes[0].color5},
        ]}
      /> : ''}
      {palettes[1] ? <Palette
        key={palettes[1].color1}
        name={palettes[1].title}
        colors={[
          {color: palettes[1].color1},
          {color: palettes[1].color2},
          {color: palettes[1].color3},
          {color: palettes[1].color4},
          {color: palettes[1].color5},
        ]}
      /> : ''}
      {palettes[2] ? <Palette
        key={palettes[2].color1}
        name={palettes[2].title}
        colors={[
          {color: palettes[2].color1},
          {color: palettes[2].color2},
          {color: palettes[2].color3},
          {color: palettes[2].color4},
          {color: palettes[2].color5},
        ]}
      /> : ''}
    </>
  )
}

export default Project;
