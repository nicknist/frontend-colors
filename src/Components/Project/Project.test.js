import React from 'react';
import Project from './Project';
import { shallow } from 'enzyme';
import { deleteProject, getPalette } from '../../apiCalls';
jest.mock('../../apiCalls');

describe('Project', () => {
  let mockInfo;
  let wrapper;
  let mockPalette;

  beforeEach(() => {
    mockPalette = {
      title: 'waddup'
    }
    deleteProject.mockImplementation(() => {});
    getPalette.mockImplementation(() => mockPalette)
    mockInfo = {
      name: 'colin',
      palette1_id: 1,
      palette2_id: 2,
      palette3_id: 3,
    }
    wrapper = shallow(<Project info={mockInfo}/>);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should run deleteProject when delete-project button is clicked', () => {
    wrapper.find('.delete-project').simulate('click');
    expect(deleteProject).toHaveBeenCalled();
  });
})
