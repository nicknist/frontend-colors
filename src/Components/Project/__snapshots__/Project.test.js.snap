// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project should match the snapshot 1`] = `
<Fragment>
  <h1 />
  <button
    className="randomizer delete-project"
    onClick={[Function]}
  >
    NUKE IT
  </button>
</Fragment>
`;
